rueppellii-lvlup-asp.net-core/Utility/ApplicationProfile.cs

```diff
@@ -34,6 +36,17 @@ public ApplicationProfile()
                 .ReverseMap();
             CreateMap<Review, ReviewerDto>()
                 .ForMember(dto => dto.Name, opt => opt.MapFrom(src => src.User.Name));
+            CreateMap<ClaimsPrincipal, User>()
+                .ForMember(user => user.Name, opt => opt.MapFrom(principal =>
+                    principal.Claims.First(claim => claim.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name").Value))
+                .ForMember(user => user.Email, opt => opt.MapFrom(principal =>
+                    principal.Claims.First(claim => claim.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress").Value))
+                .ForMember(user => user.Pic, opt => opt.MapFrom(principal =>
+                    principal.Claims.FirstOrDefault(claim => claim.Type == "profilePic").Value))
+                .ForMember(user => user.Surname, opt => opt.MapFrom(principal =>
+                    principal.Claims.FirstOrDefault(claim => claim.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname").Value))
+                .ForMember(user => user.GivenName, opt => opt.MapFrom(principal =>
+                    principal.Claims.FirstOrDefault(claim => claim.Type == "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname").Value));
```

Comment:
For magic strings like this, it's a better practice to have some static collections or configpart, where you store each and everyone of them, and then reffer them as a variable, it's better especially when you want to change your code in the future